.github/workflows/link-checker.yml

```diff
@@ -0,0 +1,43 @@
+name: Link Checker Workflow
+
+on:
+  push
+
+jobs:
+  link-checker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+
+      - name: Cache node modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install Dependencies
+        run: npm install
+
+      - name: Build
```

Comment:
So this is a bit of a naive approach: we're fully building the docs on every CI run so that we can validate all of the links in the documentation as it is published are correct.

Otherwise, I couldn't figure out how to make `lychee` validate links that are dynamically generated by Jekyll, such as https://github.com/cloud-gov/cg-site/blame/6beed2b92e5e8792d70f92190738fc2ad15cfbb1/_docs/overview.md#L13.

Right now it's only taking around 30 seconds to build the site in CI, so it seemed acceptable to me for the benefit of checking all the links on the site.